settings/blt.settings.php

```diff
@@ -64,10 +64,7 @@
 }
 
 $repo_root = dirname(DRUPAL_ROOT);
-
-if (!isset($site_path)) {
-  $site_path = \Drupal::service('site.path');
-}
+/** @var $site_path **/
```

Comment:
Well it's odd to use a variable that is not defined in this file. Where does it come from ? Should we not check for it?

Reply:
Drupal provides in `DrupalKernel::initializeSettings` -- https://git.drupalcode.org/project/drupal/blob/8.8.x/core/lib/Drupal/Core/DrupalKernel.php#L1081

Reply:
Yeah I agree it's weird. I know core defines `$site_path` as Grant pointed out, but I don't understand how it becomes exposed to the global namespace in settings files. But clearly somehow it does, or we'd have hundreds of ACSF customers complaining about this code causing a WSOD.

Reply:
Just to be super safe, I changed it to check if `$site_path` is defined first. Although it _must_ be defined, because even core's default.settings.php uses `$site_path` without checking if it's set first.

Reply:
The call stack looks roughly like:

`Index.php -> Drupal\Core\DrupalKernel::handle -> Drupal\Core\DrupalKernel::initializeSettings -> Drupal\Core\Site\Settings::initialize -> [require()](https://git.drupalcode.org/project/drupal/blob/8.8.x/core/lib/Drupal/Core/Site/Settings.php#L125)`

`$site_path` is in scope at when `settings.php` is called.

Reply:
Makes sense, thanks for the explanation!